src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -307,12 +307,12 @@ private void CreateNodeCaches()
 
             _genericReadyToRunHelpersFromDict = new NodeCache<ReadyToRunGenericHelperKey, ISymbolNode>(data =>
             {
-                return new ReadyToRunGenericLookupFromDictionaryNode(this, data.HelperId, data.Target, data.DictionaryOwner);
+                return CreateGenericLookupFromDictionaryNode(this, data.HelperId, data.Target, data.DictionaryOwner);
             });
 
             _genericReadyToRunHelpersFromType = new NodeCache<ReadyToRunGenericHelperKey, ISymbolNode>(data =>
             {
-                return new ReadyToRunGenericLookupFromTypeNode(this, data.HelperId, data.Target, data.DictionaryOwner);
+                return CreateGenericLookupFromTypeNode(this, data.HelperId, data.Target, data.DictionaryOwner);
```

Comment:
This pattern leaves us with an unnecessary virtual call. Could you follow what we do for e.g. entry points or unboxing stubs 65 lines above?

Reply:
Good point, thanks; will do.